tests/network/sriov.go

```diff
@@ -647,6 +646,40 @@ func checkInterfacesInGuest(vmi *v1.VirtualMachineInstance, interfaces []string)
 	}
 }
 
+// isInterfaceOnRootPCIComplex checks whether device is on root complex
+// Count the number of occurrences of PCI addresses along the device path
+// If only one is found, it indicates that the device is on the root complex,
+// otherwise a PCI bridge address will be listed along the path as the parent
+// of this device.
+// Examples:
+// on root       /sys/devices/pci0000:00/0000:00:03.0/net/eth1
+// not on root   /sys/devices/pci0000:00/0000:00:02.7/0000:08:00.0/net/eth1
+func isInterfaceOnRootPCIComplex(vmi *v1.VirtualMachineInstance, iface string) (bool, error) {
```

Comment:
Please consider the following alternative:
1. Get the parent PCI device of the desired NIC:
```bash
dirname $(readlink -f /sys/class/net/<NIC name>/device)
```
2. Compare the output to `/sys/devices/pci0000:00` - if its equal - the device should be on the root complex, else - its not.

I think its simpler to understand and should be potentially faster because:
1. A search operation in the [sysfs](https://en.wikipedia.org/wiki/Sysfs) is not required.
2. Regex are not used.

Reply:
Please note that the NIC's PCI address full path is given by:
```shell
readlink -f /sys/class/net/wlp1s0/device
/sys/devices/pci0000:00/0000:00:02.2/0000:01:00.0
```

Using `dirname` on this output, should give us the parent:
```shell
dirname $(readlink -f /sys/class/net/wlp1s0/device)
/sys/devices/pci0000:00/0000:00:02.2
```

Perhaps the NIC does not really resides directly under the root complex:
https://libvirt.org/pci-addresses.html

WDYT?